From 93ce697c0075aeb51db9b78c0d47e10213aa1885 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 6 Dec 2019 15:53:45 +0000
Subject: [PATCH] setup-disk: let user set ROOT_SIZE

allow user set rootfs size via ROOT_SIZE. For lvm it will be passed
directly to lvmcreate. For the sfdisk partitions it will be a number in
MiB.
---
 setup-disk.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup-disk.in b/setup-disk.in
index 5eb8638..b309919 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -989,7 +989,7 @@ native_disk_install_lvm() {
 
 	setup_lvm_volume_group $vgname $@ || return 1
 	setup_lvm_swap $vgname
-	lvcreate --quiet -n ${root_dev##*/} -l 100%FREE $vgname
+	lvcreate --quiet -n ${root_dev##*/} -l ${ROOT_SIZE:-100%FREE} $vgname
 	rc-update add lvm boot
 	setup_root $root_dev $BOOT_DEV
 }
@@ -1002,7 +1002,7 @@ native_disk_install() {
 	local prep_size=8
 	local boot_size=${BOOT_SIZE:-100}
 	local swap_size=${SWAP_SIZE}
-	local root_size=
+	local root_size=${ROOT_SIZE}
 	local root_dev= boot_dev= swap_dev=
 	init_progs $(select_bootloader) || return 1
 	confirm_erase $@ || return 1
-- 
2.24.0

